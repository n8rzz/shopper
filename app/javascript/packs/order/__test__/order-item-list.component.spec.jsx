import React from 'react';
import sinon from 'sinon';
import { shallow } from 'enzyme';
import { OrderItemList } from '../order-item-list.component';
import {
    assemblyMapMock,
    csrfMock,
    departmentMapMock,
    orderItemListMock,
    itemMapMock,
} from '../../__mocks__/mocks';

describe('OrderItemList', () => {
    test('renders correctly with valid props', () => {
        const component = shallow(<OrderItemList
            assemblyMap={assemblyMapMock}
            csrf={csrfMock}
            departmentMap={departmentMapMock}
            itemMap={itemMapMock}
            orderItems={orderItemListMock}
            onClickIsPickedHandler={sinon.spy()}
            onClickRemoveItemHandler={sinon.spy()}
        />);

        expect(component).toMatchSnapshot();
    });
});
