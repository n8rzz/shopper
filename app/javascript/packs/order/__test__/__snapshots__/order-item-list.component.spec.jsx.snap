// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderItemList renders correctly with valid props 1`] = `
<ul
  className="vlist"
>
  <OrderItem
    assemblyName="Souvlaki"
    csrf="3yIq75tyV791TFkzSZo/S4b+TVa/N5J3dI14PzMUPviGViEGeDSnGnvNix+MaB0RB0XuV7bpX8I3SmIEAr4/zQ=="
    departmentName="Bakery"
    editItemUrl="/item/15/edit"
    isPicked={false}
    itemName="Agave Syrup"
    key="orderItem-0"
    onClickIsPickedHandler={[Function]}
    onClickRemoveItemHandler={[Function]}
    orderItemId={22}
    qty={3}
  />
  <OrderItem
    assemblyName=""
    csrf="3yIq75tyV791TFkzSZo/S4b+TVa/N5J3dI14PzMUPviGViEGeDSnGnvNix+MaB0RB0XuV7bpX8I3SmIEAr4/zQ=="
    departmentName="Meat"
    editItemUrl="/item/13/edit"
    isPicked={false}
    itemName="Agave Syrup"
    key="orderItem-1"
    onClickIsPickedHandler={[Function]}
    onClickRemoveItemHandler={[Function]}
    orderItemId={23}
    qty={1}
  />
</ul>
`;
